EfYouTests/EntityServices/EntityServiceOfTTests.cs

```diff
@@ -689,33 +689,33 @@ public void Delete_ListOfIds_CallsCascadeDeleteWithFilteredEntities()
                 new DummyEntity {Id = 3},
                 new DummyEntity {Id = 4}
             };
-            var accessibleIds = new List<long> {3};
+            var accessibleIds = new List<dynamic> { 3 };
             SetMockData(entities);
-            _filterService.Setup(x => x.FilterResultsOnGet(It.IsAny<IQueryable<DummyEntity>>(), It.IsAny<List<long>>()))
-                .Returns<IQueryable<DummyEntity>, List<long>>((x, y) => x.Where(z => accessibleIds.Contains(z.Id)));
+            _filterService.Setup(x => x.FilterResultsOnGet(It.IsAny<IQueryable<DummyEntity>>(), It.IsAny<List<dynamic>>()))
+                .Returns<IQueryable<DummyEntity>, List<dynamic>>((x, y) => x.Where(z => accessibleIds.Contains(z.Id)));
 
             // Act
-            _entityService.Delete(new List<long> {3, 4});
+            _entityService.Delete(new List<dynamic> { 3, 4 });
 
             // Assert
-            _cascadeDeleteService.Verify(x => x.CascadeDelete(It.Is<List<long>>(y => y.Single() == 3)));
+            _cascadeDeleteService.Verify(x => x.CascadeDelete(It.Is<List<dynamic>>(y => y.Count == 1)));
```

Comment:
Why has this changed? Wouldn't the previous iteration work better here? I assume it's checking that the item with Id 3 still exists.

Reply:
Agree.... this is unfortunately a quirk of `dynamic` in c#.

Compiler complains when you do
`_cascadeDeleteService.Verify(x => x.CascadeDelete(It.Is<List<dynamic>>(y => y.Single() == 3)))`
and I couldn't find any way around it

Reply:
Just for my own peace of mind, you can't do a cast to dynamic on this one? So `y.Single() == (dynamic) 3`?